code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -218,54 +218,73 @@
 	glass_icon_state  = "glass_clear"
 	glass_name = "glass of holy water"
 	glass_desc = "A glass of holy water."
+	metabolization_rate = 0.4 //a fixed consumption rate to prevent balancing going out of whack
 	self_consuming = TRUE //divine intervention won't be limited by the lack of a liver
+	var/seizure = FALSE //has Nar'Sie chewed you out during this deconversion process yet?
 	ph = 7.5 //God is alkaline
 
 /datum/reagent/water/holywater/on_mob_metabolize(mob/living/L)
 	..()
 	ADD_TRAIT(L, TRAIT_HOLY, type)
+	if(iscultist(L) && !HAS_TRAIT(L, TRAIT_CULT_GHOST))
+		to_chat(L, "<span class='userdanger'>A vile holiness begins to spread its shining tendrils through your mind, purging the Geometer of Blood's influence!</span>")
 
 /datum/reagent/water/holywater/on_mob_end_metabolize(mob/living/L)
 	REMOVE_TRAIT(L, TRAIT_HOLY, type)
 	..()
 
 /datum/reagent/water/holywater/expose_mob(mob/living/exposed_mob, methods=TOUCH, reac_volume)
 	. = ..()
-	if(iscultist(exposed_mob))
-		to_chat(exposed_mob, "<span class='userdanger'>A vile holiness begins to spread its shining tendrils through your mind, purging the Geometer of Blood's influence!</span>")
+	if(HAS_TRAIT(exposed_mob, TRAIT_CULT_GHOST))
+		exposed_mob.adjustFireLoss(reac_volume, 0) //a bit extreme, perhaps, but cult ghosts are meant to be expendable and their bodies are MADE of unholy magic
+		to_chat(exposed_mob, "<span class='userdanger'>IT BURNS! IT BUUUUUURNS!</span>")
+		if((methods & TOUCH) && reac_volume >= 70 && exposed_mob.health <= exposed_mob.crit_threshold) //yo holy shit he dead
+			exposed_mob.say("AAAHHH! YOU CURSED BRAT! LOOK WHAT YOU'VE DONE! I'm MELTING! Melting! Ohhh, what a world, what a world! Who would have thought a good little spaceman like you could destroy my beautiful wickedness?! AUUUUGH! I'm gone! I'm gone! I'm going... Augh... Aaauuggghhh...", forced = "holy water") //you can theoretically say this and live using something that lets you avoid falling into crit, but eh, I'm not gonna bother to add a bunch of code for those edge cases for such a niche occurrence
```

Comment:
Maybe just shorten this to the second 'I'm melting!' or else it might get very...obstructive.

Reply:
I could make it be a 10% chance to say the full thing and a 90% (mutually exclusive) chance to say the shortened version

Reply:
What would it take for you to get a reac volume of 70u though?

Reply:
i agree that this is seems excessive but maybe with 70 units its not likely to happen

Reply:
> What would it take for you to get a reac volume of 70u though?

it has to be a reac_volume of 70u in one go, so you'd have to splash a bucket's worth of more of holy water on them